### PR TITLE
Support all resolution types

### PR DESCRIPTION
**Pull request by avnerbraverman**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -125,11 +125,8 @@ export class EIDRConnector extends BaseConnector {
 
     function validate(opt: string): string {
       if (typeof options[opt] !== 'string' || options[opt].length === 0) {
-        throw new EIDRError(
-          'Invalid ${opt}',
-          500,
-          `Invalid ${opt}: ${options[opt]}`
-        )
+        throw new EIDRError('Invalid ${opt}', 500,
+          `Invalid ${opt}: ${options[opt]}`)
```
Comment (foundingnimo) [inline src/index.ts]: Why un-prettify (here and many places..)

File: src/index.ts [hunk 2]
```diff
@@ -273,24 +261,22 @@ export class EIDRConnector extends BaseConnector {
...
   public async resolve(id: string, type = 'Full') {
     if (
+      type !== 'AlternateIDs' &&
+      type !== 'DOIKernel' &&
       type !== 'Full' &&
+      type !== 'LinkedAlternateIDs' &&
+      type !== 'Provenance' &&
       type !== 'SelfDefined' &&
-      type !== 'LinkedAlternateIDs'
+      type !== 'Simple'
```
Comment (foundingnimo) [inline src/index.ts:305]: Consider using
`if (!['AlternateIDs','DOIKernel','Full','LinkedAlternateIDs','Provenance','SelfDefined','Simple'].includes(type))`

File: src/index.ts [hunk 3]
```diff
@@ -302,38 +288,42 @@ export class EIDRConnector extends BaseConnector {
...
-        !res.LinkedAlternateIDs.LinkedAlternateID
-      ) {
-        throw new EIDRError(
-          'Unrecognized response',
-          500,
-          `Unrecognized response resolving: id=${id} type=${type}`,
-        )
+    if (type === 'AlternateIDs' || type === 'LinkedAlternateIDs') {
+      const prop = type.slice(0, -1)
+      if (!res[type] || !res[type][prop]) {
```
Comment (foundingnimo) [inline src/index.ts:345]: Is the data structure really 
`res['AlternateIDs']['AlternateID']` ?

File: src/index.ts [hunk 4]
```diff
@@ -302,38 +288,42 @@ export class EIDRConnector extends BaseConnector {
...
+          `Unrecognized response resolving: id=${id} type=${type}`)
       }
-      return res.LinkedAlternateIDs.LinkedAlternateID
+      return {
+        ID: res[type].ID,
+        [prop]: res[type][prop],
+      }
+    }
+
+    // type === 'Simple'|| type === 'Provenance' || type === 'DOIKernel'
```
Comment (foundingnimo) [inline src/index.ts:358]: SelfDefined? Full?

**Review by foundingnimo — APPROVED** [review]
Prettify no?

File: src/index.ts (hunk 4 above)
Comment (avnerbraverman) [reply src/index.ts:358]: At the top

File: src/index.ts (hunk 2 above)
Comment (avnerbraverman) [reply src/index.ts:305]: I though about that, but the "&&"s seemed more readable

File: src/index.ts (hunk 1 above)
Comment (avnerbraverman) [reply src/index.ts]: It was very verbose

**Pull request merged**